CMakeLists.txt

```diff
@@ -19,6 +19,7 @@ option(ENABLE_JEMALLOC "Enables jemalloc allocator" TRUE)
 option(ENABLE_OFFLINE_TELEMETRY "Enables FEX offline telemetry" TRUE)
 option(ENABLE_COMPILE_TIME_TRACE "Enables time trace compile option" FALSE)
 option(ENABLE_LIBCXX "Enables LLVM libc++" FALSE)
+option(ENABLE_CROSS_COMPILE "Enables cross compile FEX to aarch64 on x86_64" FALSE)
```

Comment:
The checks added below seem a bit hacky, and `ENABLE_CROSS_COMPILE` is a too generic concept here imo. It could equally mean cross-compiling from `aarch64` to `x86_64`, or cross-compiling from `aarch64-linux` to `aarch64-android`.

How about using an override architecture instead? I.e. if you wanted to cross-compile, you'd set a variable `FEX_TARGET_SYSTEM_PROCESSOR` that gets used instead of `CMAKE_SYSTEM_PROCESSOR` throughout our build scripts. If we aren't cross-compiling, the new variable can just default to `CMAKE_SYSTEM_PROCESSOR` then.

Reply:
Yes, it is one hack option, use custom target architecture looks better.